### ISSUE TITLE
Unable to start Autojack/Jack server on Ubuntu Studio 20.10

### ISSUE THREAD

**Issue by kaerumy** [ovenwerks/studio-controls#4]
jackdbus will hang on startup at the following when trying to start Jack from the GUI
```
Traceback (most recent call last):
  File "/usr/bin/studio-controls", line 479, in check_jack_status
    self.refresh_pulse_io()
  File "/usr/bin/studio-controls", line 563, in refresh_pulse_io
    active_in = self.p_in_con[self.i_in]
IndexError: list index out of range
Starting Autojack...
```
pulseaudio 1:13.99.2-1ubuntu1 
jackd  5+nmu1

**Comment by kaerumy** [comment]
jackdbus provides the following messages when trying to start Autojack from Studio Control GUI

```
Sat Oct 17 14:02:15 2020: ERROR: JackEngine::XRun: client = pulse_in-1 was not finished, state = Triggered
Sat Oct 17 14:02:15 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sat Oct 17 14:02:15 2020: ERROR: JackEngine::XRun: client = pulse_in-1 was not finished, state = Triggered
Sat Oct 17 14:02:15 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
```

**Comment by ovenwerks** [comment]
Please indicate which version of Studio-controls you are using. there have been a number while 20.10 was in beta.
Please set debug level to extra and include a copy of ~/.log/autojack.log when this happens.
A copy of ~/.config/autojack/autojackrc would also be helpful.
The problem in your first comment appears to be something that has since been fixed (studio-controls is at 2.0.8 as I recall) you would need to use apt search studio-controls to find your current installed version. It does look like removing ~/.config/autojack/autojackrc and starting over would fix this anyway. I will look at having studio-controls catch such a condition and deal with it correctly.

**Comment by kaerumy** [comment]
Issue is resolved by removing custom ~/.pulse configuration files that load up jack module sinks.

The pointer to old configuration files gave me a heads up. It looks like my legacy pulseaudio modules configuration for module-jack-sink and module-jack-source was the problem, removing this solves this issue. Previously with ubuntu-studio-controls, it looks like this file was ignored when pulseaudio with jack sinks is started.

Included is my problematic custom pulse configuration and the debug log for when this hangs, in case it might be useful.

Thank you for your work on tool, it's really simplified setup of jack and pulseaudio, and removed the need for custom configs like the source of this problem.

~/.pulse/default.pa

```
load-module module-native-protocol-unix
load-module module-jack-sink channels=2
load-module module-jack-source channels=2
load-module module-null-sink
load-module module-stream-restore
load-module module-rescue-streams
load-module module-always-sink
load-module module-suspend-on-idle
```

studio-controls 2.0.9-0ubuntu1

```
[DEFAULT]
jack = True
driver = alsa
chan-in = 0
chan-out = 0
rate = 48000
frame = 128
period = 2
zframe = 64
xdev = STX,0,0
pulse-in = pulse_in-1
pulse-out = pulse_out-1
pj-in-con = system:capture_1
pj-out-con = system:playback_1
a2j = False
dev = 0,0,0
usbauto = False
usb-single = True
usbdev = M4,0,0
log-level = 19
blacklist = C525,0,0
phone-action = switch
phone-device = 0,0,0
monitor = system:playback_1

[TABLET]
device = none
button1 = none none none
button2 = none none none
button3 = none none none
button8 = none none none
button9 = none none none
button10 = none none none
button11 = none none none
button12 = none none none
button13 = none none none
button14 = none none none
button15 = none none none
button16 = none none none
position = 0 0 2560 1440
pen = none
```

autojack.log

```
2020-10-19 10:02:42,656 - root - INFO - Autojack started: logging started
2020-10-19 10:02:42,657 - root - DEBUG - Checking for phones plugged in
2020-10-19 10:02:42,662 - root - WARNING - Phones device: 0,0,0 not found
2020-10-19 10:02:42,663 - root - INFO - Running: config_start()
2020-10-19 10:02:42,663 - root - DEBUG - log level: 10
2020-10-19 10:02:44,672 - root - DEBUG - Kill old Procs:
2020-10-19 10:02:44,678 - root - DEBUG - remove jackdbus_detect: Failed to unload module: Module module-jackdbus-detect not loaded
2020-10-19 10:02:44,691 - root - DEBUG - remove module-udev-detect: Failed to unload module: Module module-udev-detect not loaded
2020-10-19 10:02:44,697 - root - DEBUG - remove module-alsa-card: Failed to unload module: Module module-alsa-card not loaded
2020-10-19 10:02:44,698 - root - DEBUG - Found previous jack config removing
2020-10-19 10:02:44,743 - root - DEBUG - Jackdbus: --- driver select "alsa"
2020-10-19 10:02:44,779 - root - DEBUG - Jackdbus: --- driver param set "device" -> "hw:M4,0,0"
--- driver param set "rate" -> "48000"
2020-10-19 10:02:44,914 - root - DEBUG - Jackdbus: --- driver param set "period" -> "128"
--- driver param set "nperiods" -> "2"
--- start
2020-10-19 10:02:46,882 - root - WARNING - Got signal number: 2 - Dying.
2020-10-19 10:02:46,889 - root - DEBUG - Kill jack and friends: jackd: no process found
a2jmidid: no process found
2020-10-19 10:02:46,895 - root - DEBUG - Restart PA:
2020-10-19 10:03:12,398 - root - INFO - Autojack started: logging started
2020-10-19 10:03:12,398 - root - DEBUG - Checking for phones plugged in
2020-10-19 10:03:12,402 - root - WARNING - Phones device: 0,0,0 not found
2020-10-19 10:03:12,402 - root - INFO - Running: config_start()
2020-10-19 10:03:12,403 - root - DEBUG - log level: 10
2020-10-19 10:03:14,412 - root - DEBUG - Kill old Procs:
2020-10-19 10:03:14,418 - root - DEBUG - remove jackdbus_detect: Failed to unload module: Module module-jackdbus-detect not loaded
2020-10-19 10:03:14,433 - root - DEBUG - remove module-udev-detect: Failed to unload module: Module module-udev-detect not loaded
2020-10-19 10:03:14,439 - root - DEBUG - remove module-alsa-card: Failed to unload module: Module module-alsa-card not loaded
2020-10-19 10:03:14,440 - root - DEBUG - Found previous jack config removing
2020-10-19 10:03:14,483 - root - DEBUG - Jackdbus: --- driver select "alsa"
2020-10-19 10:03:14,518 - root - DEBUG - Jackdbus: --- driver param set "device" -> "hw:M4,0,0"
--- driver param set "rate" -> "48000"
2020-10-19 10:03:14,648 - root - DEBUG - Jackdbus: --- driver param set "period" -> "128"
--- driver param set "nperiods" -> "2"
--- start
2020-10-19 10:03:16,666 - root - DEBUG - jack client created
2020-10-19 10:03:17,806 - root - DEBUG - Pulseaudio: 8
2020-10-19 10:03:32,534 - root - WARNING - Got signal number: 2 - Dying.
2020-10-19 10:03:32,543 - root - DEBUG - Kill jack and friends: jackd: no process found
a2jmidid: no process found
2020-10-19 10:03:32,548 - root - DEBUG - Restart PA:

```
